std/array.d

```diff
@@ -3927,3 +3927,213 @@ unittest
     assert(appS.data == "hellow");
     assert(appA.data == "hellow");
 }
+
+/++
+Params: a = The input elements
+
+Returns: A static array constructed from `a`.
+
+The type of elements can be specified implicitly (`int[2] a = [1,2].asStatic;`)
+or explicitly (`float[2] a = [1,2].asStatic!float`). When `a` is a range,
+the number of elements has to be given as template argument (eg `2.iota.asStatic!2`).
+Size and type can be combined (eg: `2.iota.asStatic!(byte[2])`).
```

Comment:
Is there any use case in which `3.iota.asStatic` would be preferred over `asStatic!(3.iota)`?

Reply:
when using a runtime range `r` we can't use `asStatic!r` ; I've changed document unittest to use:
```
    auto input = 2.iota;
    auto a = input.asStatic!2;
```
instead of
```
    auto a = 2.iota.asStatic!2;
```

Reply:
Oh I was asking this out of the perspective of a potential reader as the documentation doesn't make this clear.

I would do something like "if the input range is known at compile-time ... can be used. If the range content is dynamic and not known at compile-time, the size needs to be set explicitly with e.g. ..."

Reply:
done (reworded ddoc)